src/UbiMesh.h

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright (c) 2013-2019 Ubidots.
+Permission is hereby granted, free of charge, to any person obtaining
+a copy of this software and associated documentation files (the
+"Software"), to deal in the Software without restriction, including
+without limitation the rights to use, copy, modify, merge, publish,
+distribute, sublicense, and/or sell copies of the Software, and to
+permit persons to whom the Software is furnished to do so, subject to
+the following conditions:
+The above copyright notice and this permission notice shall be
+included in all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+Developed and maintained by Jose Garcia for IoT Services Inc
+@jotathebest at github: https://github.com/jotathebest
+*/
+
+#ifndef _UbiMesh_H_
+#define _UbiMesh_H_
+
+#include <iterator>
+#include <map>
+#include "Particle.h"
+#include "UbiConstants.h"
+#include "UbiProtocol.h"
+#include "UbiTypes.h"
+#include "string.h"
+
+typedef std::map<uint8_t, char*> meshMap;
+static IotProtocol iotProtocolMesh = UBI_UDP;
+static char* _tokenMesh;
+static bool _debugMesh = false;
+static unsigned long time_now = millis();
+
+class UbiMesh {
+ public:
+  explicit UbiMesh(char* token);
+  void add(const char* variable_label, float value);
+  void add(const char* variable_label, float value, const char* context);
+  void add(const char* variable_label, float value, const char* context,
+           long unsigned dot_timestamp_seconds);
+  void add(const char* variable_label, float value, const char* context,
+           long unsigned dot_timestamp_seconds,
+           unsigned int dot_timestamp_millis);
+  bool meshPublish(const char* channel, const char* data);
+  bool meshPublish(const char* channel, const char* data, int timeout);
+  bool meshPublishToUbidots();
+  bool meshPublishToUbidots(const char* device_label);
+  bool meshPublishToUbidots(const char* device_label, const char* device_name);
+  void setCloudProtocol(IotProtocol iotProtocol);
+  void buildDots(std::map<uint8_t, char*>& meshMap, MeshUbi* dots);
+#if PLATFORM_ID != PLATFORM_XENON && PLATFORM_ID != PLATFORM_XENON_SOM
```

Comment:
This condition is not the same in the CPP file. could be possible that the isThrottled method is not declared in the .h file and used in the .cpp file.?

Reply:
At Ubidots.h I make that this class can be imported just for Mesh kind devices, but this method can be used only by the Boron or the Argon, that is the reason of this additional guard.